src/change-observer.js

```diff
@@ -61,20 +64,69 @@ class ChangeObserver {
    * @return {function} unwatch function
    */
   onChange(getter, handler) {
+    var entry
+    var unwatch = () => {
+      // TODO: untrack from change emitter
+      var ind = this.__observers.indexOf(entry)
+      if (ind > -1) {
+        this.__observers.splice(ind, 1)
+      }
+    }
+
     // TODO: make observers a map of <Getter> => { handlers }
-    var entry = {
+    entry = {
       getter: getter,
       handler: handler,
+      unwatchFn: unwatch,
     }
+
     this.__observers.push(entry)
+
     // return unwatch function
-    return () => {
-      // TODO: untrack from change emitter
-      var ind = this.__observers.indexOf(entry)
-      if (ind > -1) {
-        this.__observers.splice(ind, 1)
+    return unwatch
+  }
+
+  /**
+   * Calls the unwatchFn for each change observer associated with the passed in
+   * keypath or getter. If a handler function is passed as well, then the
+   * unwatchFn is called only for the change observer that holds that handler.
+   *
+   * UnwatchFns are called up the dependency chain.
+   *
+   * @param {KeyPath|Getter} getter
+   * @param {function} handler
+   */
+  unwatch(getter, handler) {
+
+    // Returns unwatchFn if no handler, or if handler is the observer's handler
+    // Otherwise returns `null`
+    var unwatchIfShould = entry => {
+      if (!handler) {
+        return entry.unwatchFn
+      }
+
+      if (handler === entry.handler) {
+        return entry.unwatchFn
       }
+
+      return null
     }
+
+    var isKeyPath = KeyPath.isKeyPath(getter)
+
+    // Collects all the unwatchFns that need to be called, without invoking
+    // them, so as to not mutate the observers collection, then invokes each.
+    each(this.__observers.map(entry => {
+      if (isKeyPath && Getter.wasKeyPath(entry.getter)) {
```

Comment:
Checking on whether both the passed getter is a keypath and if the entry's registered getter was originally a keypath, in which case we can access the original keypath using `entry.getter[0]`, since keypaths are transformed to getters using `[keypath, identity]`